src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -61,6 +61,17 @@ public async Task<int> ExecuteCommand(PackageReferenceArgs packageReferenceArgs,
                 return 0;
             }
 
+            // Convert relative path to absolute path if there is any
+            if (packageReferenceArgs.Sources?.Any() == true)
+            {
+                var projectDirectory = Path.GetDirectoryName(packageReferenceArgs.ProjectPath);
+
+                for (int i = 0; i < packageReferenceArgs.Sources.Length; i++)
+                {
+                    packageReferenceArgs.Sources[i] = UriUtility.GetAbsolutePath(projectDirectory, packageReferenceArgs.Sources[i]);
```

Comment:
Personally I think we should avoid reassigning to the args.

The `args` would ideally be an immutable group of options/requests, ideally something that's reproducible.

Reply:
created new variable.